packages/NodeTypeResolver/src/StaticTypeMapper.php

```diff
@@ -645,7 +646,11 @@ public function mapPHPStanPhpDocTypeNodeToPHPStanType(TypeNode $typeNode, Node $
         if ($typeNode instanceof GenericTypeNode) {
             if ($typeNode->type instanceof IdentifierTypeNode) {
                 $typeName = $typeNode->type->name;
-                if (in_array($typeName, ['array', 'iterable'], true)) {
+
+                // remove extra prefix
+                $typeName = ltrim($typeName, '\\');
+
+                if (in_array($typeName, ['array', 'iterable', 'Traversable'], true)) {
```

Comment:
What exactly is this condition for? There are more types like these - Iterator, Generator, ext-ds types, Doctrine collections, some SPL types... trying to white-list them for some special handling doesn't make much sense to me.

Reply:
These all need to be listed there then. For phpdoc-parser it's just a string.

Reply:
There is no way to whitelist everything. What is the purpose of this?

Reply:
Only those that have type in object fo PHPStan.

The `mapPHPStanPhpDocTypeNodeToPHPStanType()` method converts doc type to phpstan type

Reply:
If you're able to make it simpler and more robust, that would be great

Reply:
> Only those that have type in object fo PHPStan.

Can you rephrase that? I don't understand...

Would something like `=== array || is_subclass_of(Traversable)` work?

Reply:
> Can you rephrase that? I don't understand...

Sure, this is phpdoc-parsre doc type:

```php
@var int
```

To use in the code, we need to convert it to PHPStan type:

```php
$integerType = new IntegerType();
```

That's all :)


> Would something like === array || === iterable || is_subclass_of(Traversable) work?

Hard to tell, testing needs to be done, as this is 3rd party code we convert

Reply:
Let's address in new PR